oasis/models/rules.py

```diff
@@ -2,15 +2,167 @@
+                    values.append(float(data))
+
+                features_value = {}
+                for key in config["features"]:
+                    if key in Features:
+                        features_value[key] = Features[key](values)
+
+                logger.info("[job-id:{id}][metric:{metric}] extraction features {value}, "
+                            "start to compare with rule"
+                            .format(id=sub_id(self.job.id),
+                                    metric=metric, value=features_value))
+
+                not_match_count = 0
+                for rule in rules:
+                    if rule["feature"] not in features_value:
+                        continue
+
+                    if check_is_triggered(features_value[rule["feature"]],
+                                          rule["operator"], rule["value"]):
+                        logger.info("[job-id:{id}][metric:{metric}] not match rule: {rule}"
+                                    .format(id=sub_id(self.job.id), metric=metric, rule=rule))
+                        self.callback("[job] {job}, metric: {metric} not match rule: {rule}"
+                                      .format(job=dict(self.job), metric=metric,
+                                              rule=rule), self.job.slack_channel)
+                        not_match_count = not_match_count + 1
+
+                if not_match_count == 0:
+                    logger.info("[job-id:{id}][metric:{metric}] predict OK"
+                                .format(id=sub_id(self.job.id), metric=metric))
+
+            self.event.wait(timeparse(self.cfg.model["predict_interval"]))
+
+        logger.info("[job-id:{id}][metric:{metric}] stop"
+                    .format(id=sub_id(self.job.id), metric=metric))
+
+    def close(self):
+        logger.info("[job-id:{id}] closing the job"
+                    .format(id=sub_id(self.job.id)))
+        super(Rules, self).close()
+        self.timer.cancel()
+
+    def timeout_action(self):
+        logger.info("[job-id:{id}] finish the job"
+                    .format(id=sub_id(self.job.id)))
+        super(Rules, self).close()
+
+
+class RulesConfig(Config):
+    def __init__(self, config_file, config=None):
+        super(RulesConfig, self).__init__()
+
+        self.rules = dict()
+        self._init(config_file, config)
+        self._parse_rules()
+
+    def _set_default_config(self):
+        self.model.setdefault("data_range", "10m")
+        self.model.setdefault("predict_interval", "5m")
+
+        for metric in self.metrics:
+            self.metrics[metric].setdefault("features", ["mean", "std"])
+            self.metrics[metric].setdefault("rules",
+                                            ['features["std"] > 1000'])
+
+    def _parse_rules(self):
+        for metric in self.metrics:
+            rules = self.metrics[metric].get("rules")
+            rules_list = []
+            for rule in rules:
+                index_left = rule.find("[")
```

Comment:
This logic is fragile. Can you define the grammar for rule?

Like `"feature[" feature "]" op value`?